Transfers fix/features

kleytonbp opened the pull request:
Fix for issue #184.

- Created private method _intializeTransferInstrument_ to initialize _transferInstrument_ if it doesn't exist. (_**changed, look comments bellow**_)
- Calling _intializeTransferInstrument_ in _setTransfers_ and _setHolder_. (_**changed, look comments bellow**_)
- StyleCI fix.

kleytonbp commented:
I followed the issue instructions to fix this problem, checking the existence of $this->data->transferInstrument in setHolder and setTransfers, but i was wondering if would be better only instantiate transferInstrument in initialize() method. Any thoughts about it?

caiogaspar commented:
Hi @kleytonbp, I tried to test your pull request and check if is better initialize in initialize() method then I realized that our SDK has no tests for Transfers. Would you mind to add tests to this resource?

caiogaspar commented:
#192

kleytonbp commented:
Added to this PR other changes related to Transfers.
- Added show, list and revert (issue #186)
- Added tests (issue #192)
- Added documentation (examples on README)

I couldn't test properly the features show, list and revert, because i got status 400 when i tried to execute a new transfer (api bug?). Anyway, all unit tests are passing.

About the issue #184, i tried to improve a bit the old fix. Now transferInstrument, bankAccount, holder and taxDocument are instantiated right on initialize method. I hope it don't create any problem.

kleytonbp commented:
@caiogaspar Tests added :D

joaolucasl commented on tests/Resource/TransfersTest.php:
```diff
@@ -0,0 +1,73 @@
+    {
+        $this->mockHttpSession($this->body_transfers_create);
+
+        $transfer_id = 'TRA-28HRLYNLMUFH';
```
Esse teste tá bem legal no geral, mas como tá usando um ID fixo e vc só pode consultar `transfers` que vc mesmo criou, quando se roda os testes batendo direto na API (rodamos aqui e deu pau =/)

O ideal seria criar uma transfer dentro do teste, pegar o ID dela e passar. Pode ser?

joaolucasl commented on tests/Resource/TransfersTest.php:
```diff
@@ -0,0 +1,73 @@
+    {
+        $this->mockHttpSession($this->body_transfers_revert);
+
+        $transfer_id = 'TRA-28HRLYNLMUFH';
```
Mesmo comentário do teste acima!

kleytonbp commented:
@joaolucasl Alterações realizadas 👍

kleytonbp commented:
@joaolucasl Em relação a documentação e os links gerados pelo Github, tá rolando um problema com a última sessão de "Consulta" (inclusive na versão oficial/atual de develop). Aparentemente isso acontece porque existe no índice "Consulta" em "Reembolsos", porém na documentação não existe essa sessão.
A dúvida é se essa sessão realmente não deveria existir, então retirar ela do índice pra resolver o problema, ou se ela deveria existir e precisa ser adicionada.

joaolucasl commented:
É pra ter quando a feature sair, mas se não me engano não está implementado. Pode tirar dos docs, colocamos quando for implementado! :)

joaolucasl commented:
@kleytonbp, tudo bem? Só atualizar os docs removendo aquela seção que a gente já dá merge, pode ser? A gente não consegue dar push na sua branch, senão fariamos aqui :)

kleytonbp commented:
@joaolucasl Opa, mil desculpas pela demora :) Realizei a atualização do indice.
